editor/src/components/images.ts

```diff
@@ -54,14 +48,88 @@ export function getImageSrc(
   }
 }
 
-export function parseImageMultiplier(imagePath: string): number {
-  const imageMultiplierRegex = /.*@(\d*)x\..*/
-  const imageMultiplierResult = imageMultiplierRegex.exec(imagePath)
-  let multiplier: number = 1
-  if (imageMultiplierResult != null && imageMultiplierResult.length === 2) {
-    multiplier = Number.parseInt(imageMultiplierResult[1])
+export interface FilenameParts {
+  filename: string
+  extension: string
+  multiplier?: number
+  deduplicationSeqNumber?: number
+}
+
+export function filenameFromParts(parts: FilenameParts): string {
+  const { filename, multiplier, extension, deduplicationSeqNumber } = parts
+  const multiplierString = optionalMap((m) => `@${m}x`, multiplier) ?? ''
+  const dedupeString = optionalMap((n) => `_${n}`, deduplicationSeqNumber) ?? ''
+  return `${filename}${dedupeString}${multiplierString}.${extension}`
+}
+
+export interface LastPartSeparatedByResult<T> {
+  part: T
+  rest: string
+}
+
+function lastPartSeparatedBy<Separator extends string, T>(
+  sepBy: Separator,
+  make: (_: string) => T | null,
+  s: string,
+): LastPartSeparatedByResult<T> | null {
+  const parts = s.split(sepBy)
+  if (s.length < 2) {
+    return null
+  }
+
+  const made = make(parts[parts.length - 1])
+  if (made == null) {
+    return null
+  }
+
+  return {
+    rest: parts.slice(0, -1).join(sepBy),
+    part: made,
+  }
+}
+
+const safeParseInt = (raw: string): number | null => {
+  const result = Number.parseInt(raw)
+  return isNaN(result) ? null : result
+}
```

Comment:
We have a `safeParseInt` in [`math-utils.ts`](https://github.com/concrete-utopia/utopia/blob/master/editor/src/core/shared/math-utils.ts#L915) which is truly awful, and thankfully unused. Please can you replace that version with this one

Reply:
thanks, done!